parsely/src/test/java/com/parsely/parselyandroid/InMemoryBufferTest.kt

```diff
@@ -0,0 +1,79 @@
+package com.parsely.parselyandroid
+
+import androidx.test.core.app.ApplicationProvider
+import kotlin.time.Duration.Companion.seconds
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.test.advanceTimeBy
+import kotlinx.coroutines.test.runCurrent
+import kotlinx.coroutines.test.runTest
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@RunWith(RobolectricTestRunner::class)
+internal class InMemoryBufferTest {
+
+    private lateinit var sut: InMemoryBuffer
+    private val repository = FakeLocalStorageRepository()
+
+    @Test
+    fun `when adding a new event, then save it to local storage and run onEventAdded listener`() = runTest {
```

Comment:
Suggestion (💡): Consider splitting this test into two:

1. `when adding a new event, then save it to local storage`
2. `given a listener, when adding a new event, then run listener`